src/metadataGeneration/methodGenerator.ts

```diff
@@ -40,7 +41,7 @@ export class MethodGenerator {
       nodeType = typeChecker.typeToTypeNode(implicitType) as ts.TypeNode;
     }
     const type = new TypeResolver(nodeType, this.current).resolve();
-    const responses = this.getMethodResponses();
+    const responses = this.commonResponses ? this.commonResponses.concat(this.getMethodResponses()) : this.getMethodResponses();
```

Comment:
we should always pass a common responses array here, so we don't have to check if it exists

Reply:
Agreed, fixed.